gimie/sources/common/license.py

```diff
@@ -1,34 +1,50 @@
+import os
 import re
-from spdx_license_list import LICENSES
+from spdx_license_list import LICENSES, License
 from scancode.api import get_licenses
-from typing import List
+from typing import Iterable, List, Optional
 from gimie.io import Resource, iterable_to_stream, RemoteResource
+import tempfile
 
 
-def _get_licenses(temp_file_path: str) -> str:
-    """Takes a file with a license text in it, and matches this using the scancode API to a license to get some possible
-    license matches. The highest match is then returned as a spdx license ID"""
-    license_detections = get_licenses(temp_file_path, include_text=True)[
+def _get_license_url(license_file: Resource) -> Optional[str]:
+    """Takes the path of a text file containing a license text, and matches this
+    using the scancode API to get possible license matches. The best match is
+    then returned as a spdx license URL"""
+    temp_file = tempfile.NamedTemporaryFile(delete=False)
+    temp_file.write(license_file.open().read())
+    temp_file.close()
+
+    license_detections = get_licenses(temp_file.name, include_text=True)[
         "license_detections"
     ]
-    license_id = get_license_with_highest_coverage(license_detections)
-    spdx_license_id = get_spdx_license_id(LICENSES, license_id)
+    license_id = get_license_with_highest_coverage(license_detections)  # type: ignore
+    spdx_license_id = get_spdx_license_id(LICENSES.keys(), license_id)
+    os.unlink(temp_file.name)
+    if spdx_license_id:
+        return f"https://spdx.org/licenses/{str(spdx_license_id)}.html"
+
+    return None
+
 
-    return spdx_license_id
+def get_spdx_license_id(
+    ref_licenses: Iterable[str], license_id: Optional[str]
+) -> Optional[str]:
+    """Given a scancode API license ID also known as a license detection, returns the correctly capitalized
+    spdx id corresponding to it.
 
+    Parameters
+    ----------
+    ref_licenses: Iterable[str]
```

Comment:
I don't love ref_licenses. It's compact, but I would suggest something like SPDX_Ids , since it's a constant (and it's plural). or maybe spdx_id_list.

Not very important, but it would help me understand without reading the typehint what ref_licenses is referring to.